### PR TITLE
compliance: Download tailoring file for tailored policies

### PR DESCRIPTION
**Pull request by dLobatog**
Notice this is only meant to work on CI for now - it cannot be promoted to production as hitting the endpoint for downloading tailoring policies would not work. I can change the code to handle that situation but it's probably better to hold off releasing this until the production API responds appropriately with a `policy['tailored']` field.  

:warning: To create a tailored policy, you would need:
https://github.com/RedHatInsights/compliance-frontend/pull/423 in the frontend
https://github.com/RedHatInsights/compliance-backend/pull/352 in the backend

### CODE REVIEW

**Comment by dLobatog** [comment]
@akofink They are missing on production, on CI/QA they should be available right? I do see them on my env

**Comment by akofink** [comment]
They are missing on all environments. NB: You're using the systems API endpoint, not profiles endpoint.

File: insights/client/apps/compliance/__init__.py [hunk 1]
```diff
@@ -30,22 +30,50 @@ def oscap_scan(self):
...
     # We need to update compliance-backend to fix this
     def get_policies(self):
-        response = self.conn.session.get("https://{0}/compliance/systems".format(self.config.base_url), params={'search': 'name={0}'.format(self.hostname)})
+        response = self.conn.session.get("https://{0}/compliance/profiles".format(self.config.base_url),
+                                         params={'search': 'system_names={0}'.format(self.hostname)})
+        logger.debug("Content of the response: {0} - {1}".format(response,
+                                                                 response.json()))
         if response.status_code == 200:
-            return (response.json().get('data') or [{}])[0].get('attributes', {}).get('profiles', [])
+            return (response.json().get('data') or [])
```
Comment (dLobatog) [inline insights/client/apps/compliance/__init__.py]: The newer response looks like:

```json
{
  "data": [
    {
      "id": "057320aa-a68a-4f98-b703-993c4796508e",
      "type": "profile",
      "attributes": {
        "name": "NIST National Checklist Program Security Guide",
        "ref_id": "xccdf_org.ssgproject.content_profile_ncp",
        "description": "This compliance profile reflects the core set of security\nrelated configuration settings for deployment of Red Hat Enterprise\nLinux 7.x into U.S. Defense, Intelligence, and Civilian agencies.\nDevelopment partners and sponsors include the U.S. National Institute\nof Standards and Technology (NIST), U.S. Department of Defense,\nthe National Security Agency, and Red Hat.\n\nThis baseline implements configuration requirements from the following\nsources:\n\n- Committee on National Security Systems Instruction No. 1253 (CNSSI 1253)\n- NIST Controlled Unclassified Information (NIST 800-171)\n- NIST 800-53 control selections for MODERATE impact systems (NIST 800-53)\n- U.S. Government Configuration Baseline (USGCB)\n- NIAP Protection Profile for General Purpose Operating Systems v4.2.1 (OSPP v4.2.1)\n- DISA Operating System Security Requirements Guide (OS SRG)\n\nFor any differing configuration requirements, e.g. password lengths, the stricter\nsecurity setting was chosen. Security Requirement Traceability Guides (RTMs) and\nsample System Security Configuration Guides are provided via the\nscap-security-guide-docs package.\n\nThis profile reflects U.S. Government consensus content and is developed through\nthe OpenSCAP/SCAP Security Guide initiative, championed by the National\nSecurity Agency. Except for differences in formatting to accommodate\npublishing processes, this profile mirrors OpenSCAP/SCAP Security Guide\ncontent as minor divergences, such as bugfixes, work through the\nconsensus and release processes.",
        "score": "50.0",
        "parent_profile_id": null,
        "canonical": true,
        "tailored": false,
        "total_host_count": 63,
        "compliant_host_count": 0
      }
    },
    {
      "id": "1074c44f-a9bc-441a-9375-f14a1bbef894",
      "type": "profile",
      "attributes": {
        "name": "PCI-DSS v3.2.1 Control Baseline for Red Hat Enterprise Linux 8",
        "ref_id": "xccdf_org.ssgproject.content_profile_pci-dss",
        "description": "Ensures PCI-DSS v3.2.1 security configuration settings are applied.",
        "score": "50.0",
        "parent_profile_id": null,
        "canonical": true,
        "tailored": false,
        "total_host_count": 5,
        "compliant_host_count": 0
      }
    }
  ],
  "meta": {
    "total": 2,
    "search": "system_names = lobatolan-0",
    "limit": 10,
    "offset": 1
  },
  "links": {
    "first": "/api/compliance/profiles?limit=10&search=system_names = lobatolan-0&offset=1",
    "last": "/api/compliance/profiles?limit=10&search=system_names = lobatolan-0&offset=1"
  }
}{
  "data": [
    {
      "id": "057320aa-a68a-4f98-b703-993c4796508e",
      "type": "profile",
      "attributes": {
        "name": "NIST National Checklist Program Security Guide",
        "ref_id": "xccdf_org.ssgproject.content_profile_ncp",
        "description": "This compliance profile reflects the core set of security\nrelated configuration settings for deployment of Red Hat Enterprise\nLinux 7.x into U.S. Defense, Intelligence, and Civilian agencies.\nDevelopment partners and sponsors include the U.S. National Institute\nof Standards and Technology (NIST), U.S. Department of Defense,\nthe National Security Agency, and Red Hat.\n\nThis baseline implements configuration requirements from the following\nsources:\n\n- Committee on National Security Systems Instruction No. 1253 (CNSSI 1253)\n- NIST Controlled Unclassified Information (NIST 800-171)\n- NIST 800-53 control selections for MODERATE impact systems (NIST 800-53)\n- U.S. Government Configuration Baseline (USGCB)\n- NIAP Protection Profile for General Purpose Operating Systems v4.2.1 (OSPP v4.2.1)\n- DISA Operating System Security Requirements Guide (OS SRG)\n\nFor any differing configuration requirements, e.g. password lengths, the stricter\nsecurity setting was chosen. Security Requirement Traceability Guides (RTMs) and\nsample System Security Configuration Guides are provided via the\nscap-security-guide-docs package.\n\nThis profile reflects U.S. Government consensus content and is developed through\nthe OpenSCAP/SCAP Security Guide initiative, championed by the National\nSecurity Agency. Except for differences in formatting to accommodate\npublishing processes, this profile mirrors OpenSCAP/SCAP Security Guide\ncontent as minor divergences, such as bugfixes, work through the\nconsensus and release processes.",
        "score": "50.0",
        "parent_profile_id": null,
        "canonical": true,
        "tailored": false,
        "total_host_count": 63,
        "compliant_host_count": 0
      }
    },
    {
      "id": "1074c44f-a9bc-441a-9375-f14a1bbef894",
      "type": "profile",
      "attributes": {
        "name": "PCI-DSS v3.2.1 Control Baseline for Red Hat Enterprise Linux 8",
        "ref_id": "xccdf_org.ssgproject.content_profile_pci-dss",
        "description": "Ensures PCI-DSS v3.2.1 security configuration settings are applied.",
        "score": "50.0",
        "parent_profile_id": null,
        "canonical": true,
        "tailored": false,
        "total_host_count": 5,
        "compliant_host_count": 0
      }
    }
  ],
  "meta": {
    "total": 2,
    "search": "system_names = lobatolan-0",
    "limit": 10,
    "offset": 1
  },
  "links": {
    "first": "/api/compliance/profiles?limit=10&search=system_names = lobatolan-0&offset=1",
    "last": "/api/compliance/profiles?limit=10&search=system_names = lobatolan-0&offset=1"
  }
}
```

File: insights/client/auto_config.py [hunk 2]
```diff
@@ -239,5 +239,5 @@ def try_auto_configuration(config):
         if not _try_satellite6_configuration(config):
             _try_satellite5_configuration(config)
     if not config.legacy_upload and 'cloud.redhat.com' not in config.base_url:
-        config.base_url = config.base_url + '/platform'
+        config.base_url = config.base_url
```
Comment (akofink) [inline insights/client/auto_config.py]: Not intended, I guess?

File: insights/tests/client/apps/test_compliance.py [hunk 3]
```diff
@@ -123,3 +124,32 @@ def test_run_scan_fail(config, call):
     with raises(SystemExit):
         compliance_client.run_scan('ref_id', '/nonexistent', output_path)
     call.assert_called_with("oscap xccdf eval --profile ref_id --results " + output_path + ' /nonexistent', keep_rc=True, env=env)
+
+
```
Comment (akofink) [inline insights/tests/client/apps/test_compliance.py]: What about a test for malformed API response. i.e. a response missing 'attributes' or 'tailored'. This client might be run against the wrong version of our backend, so I think it'd be good to handle those cases.

File: insights/client/auto_config.py (hunk 2 above)
Comment (dLobatog) [reply insights/client/auto_config.py]: Yup, my bad - I use this to be able to use `base_url=ci.foo.redhat.com:1377`

File: insights/tests/client/apps/test_compliance.py (hunk 3 above)
Comment (dLobatog) [reply insights/tests/client/apps/test_compliance.py]: Updated, see ` test_tailored_file_is_not_downloaded_if_tailored_is_missing`

**Pull request merged**